0000-catalog.md

```diff
@@ -0,0 +1,744 @@
+
+### CatalogPayload Transaction
+
+```
+message CatalogPayload { 
+    enum Actions { 
+        UNSET_ACTION = 0; 
+        CATALOG_CREATE = 1; 
+        CATALOG_UPDATE = 2; 
+        CATALOG_DELETE = 3; 
+    }
+
+    Action action = 1;
+
+    // Approximately when transaction was submitted, as a Unix UTC timestamp
+    uint64 timestamp = 2;
+
+    CatalogCreateAction catalog_create = 3; 
+    CatalogUpdateAction catalog_update = 4; 
+    CatalogDeleteAction catalog_delete = 5; 
+} 
+```
+
+### CatalogProductPayload Transaction
+CatalogProductPayload contains an action enum and the associated action payload. 
+This allows for the action payload to be dispatched to the appropriate logic.
+Only the defined actions are available and only one action payload should be 
+defined in the CatalogProductPayload.
+
+```
+message CatalogProductPayload { 
+    enum Actions { 
+        UNSET_ACTION = 0; 
+        CATALOG_PRODUCT_CREATE = 1; 
+        CATALOG_PRODUCT_UPDATE = 2; 
+        CATALOG_PRODUCT_DELETE = 3; 
+    }
+
+    Action action = 1;
+
+    // Approximately when transaction was submitted, as a Unix UTC timestamp
+    uint64 timestamp = 2;
+
+    CatalogProductCreateAction catalog_product_create = 3; 
+    CatalogProductUpdateAction catalog_product_update = 4; 
+    CatalogProductDeleteAction catalog_product_delete = 5; 
+} 
+```
+
+```
+message CatalogProductOperationPayload { 
+    enum Actions { 
+        UNSET_ACTION = 0;
+        SET_CATALOG_PRODUCT_STATUS = 1;
+    }
+
+    Action action = 1;
+
+    // Approximately when transaction was submitted, as a Unix UTC timestamp
+    uint64 timestamp = 2;
+
+    SetCatalogProductStatusAction set_catalog_product_status = 3;  
+} 
+```
+
+## Catalog Actions
+
+### CatalogCreateAction
+CatalogCreateAction adds a new catalog to state. The transaction should be submitted 
+by an agent, which is identified by its signing key, acting on behalf of the 
+organization that corresponds to the owner in the create transaction. (Organizations 
+and agents are defined by the Pike smart contract.)
+```
+message CatalogCreateAction { 
+    // GS1 Company Prefix from owner + catalog_id are use as 
+    // a composite key for determining the state address
+    string owner = 1;
+    string catalog_id = 2;
+    string name = 3;
+    repeated PropertyValues properties = 4; 
+} 
+```
+Validation requirements:
+
+- If a catalog with catalog_id already exists the transaction is invalid.
+- The signer of the transaction must be an agent in the Pike state and must belong 
+to an organization in Pike state, otherwise the transaction is invalid.
+- The agent must have the permission can_create_catalog for the organization, 
+otherwise the transaction is invalid.
+
+If all requirements are met, the transaction will be accepted, the batch will be 
+written to a block, and the catalog will be created in state.
+
+The inputs for CatalogCreateAction must include:
+
+Address of the Agent submitting the transaction
```

Comment:
This is not formatted correctly (same with the rest of inputs/outputs)

Reply:
This is still an issue

Reply:
✅

Reply:
✅